basic_module/account/src/account/internal.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::AccountError;
+use crate::{get_context, get_storage_id, Account};
+use key::Address;
+
+pub fn add_balance(address: &Address, val: u64) {
+    let context = get_context();
```

Comment:
How about adding `val == 0` check to reduce storage access when any change is not expected like the previous `TopLevelState` operation.

Reply:
I added!